runtime/include/DataView.hpp

```diff
@@ -16,35 +16,30 @@
 
 #include <Exception.hpp>
 
-// MemRef type definition
-template <typename T, size_t R> struct MemRefT {
-    T *data_allocated;
-    T *data_aligned;
-    size_t offset;
-    size_t sizes[R];
-    size_t strides[R];
-};
-
 /**
- * A multi-dimensional view for MemRef<T, R> types.
+ * A multi-dimensional view for MemRef-like and std::vector<T> types.
  *
  * @tparam T The underlying data type
  * @tparam R The Rank (R >= 0)
  *
  * @note A forward iterator is implemented in this view for traversing over the entire
  * elements of MemRef types rank-by-rank starting from the last dimension (R-1). For example,
- * The MemRefView iterator for MemRef<T, 2> starts from index (0, 0) and traverses elements
+ * The DataView iterator for MemRef<T, 2> starts from index (0, 0) and traverses elements
  * in the following order:
  * (0, 0), ..., (0, sizes[1]-1), (1, 0), ..., (1, sizes[1]-1), ... (sizes[0]-1, sizes[1]-1).
  */
-template <typename T, size_t R> class MemRefView {
+template <typename T, size_t R> class DataView {
   private:
-    const MemRefT<T, R> *buffer;
+    T *data_aligned;
+    size_t offset;
+    size_t sizes[R] = {0};
+    size_t strides[R] = {0};
```

Comment:
Does this set the pointer value to zero when the rank is 0 or does it try to place one element (the `0`) into an array of size 0? 😅 (Does C allow arrays of size zero?)

Reply:
I supposed there's no use of R=0 and used these C-like initialization to avoid creating a unique_ptr of size_t[R] in the constructor. :smile: In general, arrays of rank 0 are valid in C/C++ and `sizes[0]` is a valid type. But the initialization will be failed as you cannot init a C-array of rank 0. So, the current code doesn't support R=0.